lms/static/scripts/frontend_apps/components/test/ContentSelector-test.js

```diff
@@ -130,48 +140,80 @@ describe('ContentSelector', () => {
     });
   });
 
-  it('shows LMS file dialog when "Select PDF from Canvas" is clicked', () => {
-    const wrapper = renderContentSelector();
+  describe('LMS file dialog', () => {
+    [
+      {
+        name: 'Canvas',
+        buttonTestId: 'canvas-file-button',
+        files: () => fakeConfig.filePicker.canvas.listFiles,
+      },
+      {
+        name: 'Blackboard',
+        buttonTestId: 'blackboard-file-button',
+        files: () => fakeConfig.filePicker.blackboard.listFiles,
+      },
+    ].forEach(test => {
+      it(`shows LMS file dialog when "Select PDF from ${test.name}" is clicked`, () => {
+        const wrapper = renderContentSelector();
 
-    assert.isFalse(isLoadingIndicatorVisible(wrapper));
+        assert.isFalse(isLoadingIndicatorVisible(wrapper));
 
-    const btn = wrapper.find('LabeledButton[data-testid="lms-file-button"]');
-    interact(wrapper, () => {
-      btn.props().onClick();
-    });
+        const btn = wrapper.find(
+          `LabeledButton[data-testid="${test.buttonTestId}"]`
+        );
+        interact(wrapper, () => {
+          btn.props().onClick();
+        });
 
-    assert.isTrue(isLoadingIndicatorVisible(wrapper));
+        assert.isTrue(isLoadingIndicatorVisible(wrapper));
 
-    const filePicker = wrapper.find('LMSFilePicker');
-    assert.isTrue(filePicker.exists());
-    assert.equal(filePicker.prop('authToken'), fakeConfig.api.authToken);
-    assert.equal(
-      filePicker.prop('listFilesApi'),
-      fakeConfig.filePicker.canvas.listFiles
-    );
+        const filePicker = wrapper.find('LMSFilePicker');
+        assert.isTrue(filePicker.exists());
+        assert.equal(filePicker.prop('authToken'), fakeConfig.api.authToken);
 
-    interact(wrapper, () => {
-      filePicker.props().onCancel();
-    });
-    assert.isFalse(isLoadingIndicatorVisible(wrapper));
-  });
-
-  it('supports selecting an LMS file', () => {
-    const onSelectContent = sinon.stub();
-    const wrapper = renderContentSelector({
-      defaultActiveDialog: 'lmsFile',
-      onSelectContent,
-    });
-    const file = { id: 123 };
+        assert.equal(filePicker.prop('listFilesApi'), test.files());
 
-    const picker = wrapper.find('LMSFilePicker');
-    interact(wrapper, () => {
-      picker.props().onSelectFile(file);
+        interact(wrapper, () => {
+          filePicker.props().onCancel();
+        });
+        assert.isFalse(isLoadingIndicatorVisible(wrapper));
+      });
     });
 
-    assert.calledWith(onSelectContent, {
-      type: 'file',
-      file,
+    [
+      {
+        name: 'canvas',
+        dialogName: 'canvasFile',
+        file: { id: 123 },
+        result: {
+          type: 'file',
+          file: { id: 123 },
+        },
+      },
+      {
+        name: 'blackboard',
+        dialogName: 'blackboardFile',
+        file: { id: 'blackboard://content-resource/123' },
```

Comment:
👍 for using "realistic" looking file IDs here in the sense that they match the shape of values from the actual app.